sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionEvalHelper.scala

```diff
@@ -225,7 +264,9 @@ trait ExpressionEvalHelper extends GeneratorDrivenPropertyChecks with PlanTestBa
           val lit = InternalRow(expected, expected)
           val expectedRow =
             UnsafeProjection.create(Array(expression.dataType, expression.dataType)).apply(lit)
-          if (unsafeRow != expectedRow) {
+          val field = StructField("field", expression.dataType)
+          val dataType = StructType(field :: field :: Nil)
+          if (!checkResult(unsafeRow, expectedRow, dataType)) {
```

Comment:
What's this for?

Reply:
```UnsafeRow```s are compared based on equality of backing arrays. This approach doesn't work well when ignoring order in unsafe representation of maps.